### PR TITLE
Updating to react 14.2

### PR DESCRIPTION
**Pull request by chrisnojima**
Tested on ios and android (but could use more eyes actually building this stuff)
@keybase/react-hackers @MarcoPolo especially for the android stuff

### CODE REVIEW

File: react-native/android/app/react.gradle [hunk 1]
```diff
@@ -0,0 +1,75 @@
+def config = project.hasProperty("react") ? project.react : [];
```
Comment (chrisnojima) [inline react-native/android/app/react.gradle]: Due to changes in the assets system both ios and android now build the bundle as part of their build process. Hence I've removed the commands from npm

File: react-native/android/app/proguard-rules.pro [hunk 2]
```diff
@@ -15,3 +15,46 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
```
Comment (chrisnojima) [inline react-native/android/app/proguard-rules.pro]: this all came from react-native's upgrade config, @MarcoPolo please sanity check this

File: react-native/ios/Keybase/Images.xcassets/service_coinbase.imageset/Contents.json [hunk 3]
```diff
@@ -1,21 +0,0 @@
-{
```
Comment (chrisnojima) [inline react-native/ios/Keybase/Images.xcassets/service_coinbase.imageset/Contents.json]: All images are now in the js land and not in the platform specific project folders

File: react-native/package.json [hunk 4]
```diff
@@ -5,8 +5,6 @@
   "scripts": {
     "start": "node_modules/react-native/packager/packager.sh",
     "pod-install": "cd ios; pod install",
-    "reactbundle-ios": "node node_modules/react-native/local-cli/cli.js bundle --minify --root react --out ios/Keybase/main.jsbundle --url \"/react/index.bundle?platform=ios&dev=\"",
```
Comment (chrisnojima) [inline react-native/package.json]: This is part of the normal build on both platforms now

File: react-native/package.json [hunk 5]
```diff
@@ -14,14 +12,14 @@
...
+    "framed-msgpack-rpc": "keybase/node-framed-msgpack-rpc#nojima/keybase-client-changes",
     "iced-runtime": "^1.0.3",
     "immutable": "^3.7.5",
     "lodash": "^3.10.1",
     "moment": "^2.10.6",
-    "purepack": "https://github.com/keybase/purepack.git#nojima/keybase-client-changes",
+    "purepack": "keybase/purepack#nojima/keybase-client-changes",
     "qrcode-generator": "^1.0.0",
-    "react-native": "git://github.com/keybase/react-native#packager-allows-mobile-files",
+    "react-native": "keybase/react-native#packager-allows-mobile-files-0-14-2",
```
Comment (chrisnojima) [inline react-native/package.json]: using the new npm format (drop the git part) and pointing to a version specific branch. we own a fork to allow the .mobile.js fallbacks in the packager

File: react-native/react/pinentry/index.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 'use strict'
 
 import React, { Component } from '../base-react'
-import PinentryRender from './pinentry-render'
+import PinentryRender from './pinentry.render'
```
Comment (chrisnojima) [inline react-native/react/pinentry/index.js]: had to fixup pinentry so mobile builds, also going to make all the names blah.render.js everywhere soon

**Comment by MarcoPolo** [comment]
Could you also update the tests to reflect the new way of building the app? since you removed the npm commands?

File: react-native/react/pinentry/index.js (hunk 6 above)
Comment (MarcoPolo) [inline react-native/react/pinentry/index.js]: wouldn't it be blah.render.mobile.js?

File: react-native/android/app/proguard-rules.pro (hunk 2 above)
Comment (MarcoPolo) [inline react-native/android/app/proguard-rules.pro]: yeah this looks fine, we weren't using progaurd yet anyways. It's for optimizing the size of the app and removing dead code.

**Comment by MarcoPolo** [comment]
this is failing circle because of memory limits, I can fix that now or in a future PR. Your call. Feel free to merge this :+1:

**Pull request merged**